### DISCUSSION TITLE
Bonsai crashing with Spinnaker: GenICam::AccessException Error

### DISCUSSION THREAD

**Discussion by srijanibiswas** [bonsai-rx/bonsai#1350] [Q&A]
I'm using Bonsai with the Neurophotometrics fp3002 system, but Bonsai keeps crashing with this error:

![spinnaker error](https://user-images.githubusercontent.com/132516870/236910372-feb7514d-304a-466c-aace-1dcd0310f2c9.png)

This happens at variable times - anything between 5 minutes to almost an hour since the start of recording. I've tried changing the usb cable, updating Bonsai drivers, and have generally looked online to troubleshoot this but have not found a solution yet. I've had this problem for months, so any help on why I could be getting this error and how to resolve it would be great!!

Here is a picture of the Bonsai layout I'm using:

![bonsai_layout_with_labels](https://user-images.githubusercontent.com/132516870/236911818-134c5a92-6247-45dd-8d4a-2167ee0016bf.png)

Please let me know if you need anything else.

Thanks!

**Comment by gracedodis** [comment]
hey! did you ever end up solving this issue? i'm dealing with the same thing now. it's either the exact same error message you were getting or double clicking the fp3002 node fails to initialize connection. here's the error message that's showing in the script as well. 
<img width="1715" height="508" alt="Screenshot 2025-12-03 184916" src="https://github.com/user-attachments/assets/b1080ead-f2c2-4b90-95d2-c5cca1a4d7cf" />

**Comment by bruno-f-cruz** [comment]
What version of the spinnaker SDK do you have installed and what is the version that the manufacturer of the device recommends?

**Reply by gracedodis** [reply]
I have the SpinnakerSDK_FULL_1.29.0.5_x64 version, which "Currently only the 1.29.05 version of the SDK is compatible" according to the Aug 2025 updated manual on the device manufacturer site. I actually resolved the problem above by completly wiping Bonsai, spinnaker, and any associated packages and reinstalling everything because I realized I had installed a newer version of spinnaker SDK. My workflow was finally working and I was so excited, but then I tried to add in a Flir camera to my workflow with a spinnaker capture node to record behavior videos during fiber recording/opto stimulation, and now that same error is popping up again. Even when I disconnect the Flir camera and remove the nodes, I'm getting an error that the "FP3002 device can not be found."

**Reply by banchan86** [reply]
Hi @gracedodis 

1) I recall there was one instance where adding a FLIR camera caused a multiple-camera found error, which was fixed in a `MBF.Neurophotometrics` package update (0.6.5). This is different from what you are seeing, but just in case, what version of the package are you using?

2) There are some common troubleshooting steps recommended for that error in the Neurophotometrics manual. Can you try that out? 
<img width="891" height="1305" alt="image" src="https://github.com/user-attachments/assets/1892c85a-28b4-4eac-bf87-f83522f260e5" />

3) You can also try reaching out directly to Neurophotometrics support for assistance with the Neurophotometrics package specifically.

**Reply by gracedodis** [reply]
Hey! Thanks for the response. You were right, it was a multiple cameras related issue. The FP3002 system has Flir cameras itself, so its common for Bonsai to be finnicky when connecting another Flir behavior camera. These are the steps I took in case anyone reading this has a similar issue:

1. Disconnect behavior camera from PC
2. Open FP3002 node
3. Close FP3002 node
4. Reconnect behavior camera

After connecting the first time, Bonsai will recognize the serial numbers and the issue shouldn't persist.
